googler

```diff
@@ -836,11 +836,12 @@ def new_connection(domain=None):
     return http.client.HTTPSConnection(domain if domain else server, timeout=45)
 
 
-def quit_program(conn):
-    """Closes a connection and quits the program"""
+def quit_program(exitcode=1):
+    """Close the connection and quits the program."""
 
+    global conn
```

Comment:
As I mentioned `fetch_results()` use the global `conn` too, so I don't see why `quit_program()` can't use it. We can make `new_connection()` handle the closure of the old one (if not `None`) and assign to the global too. I'm fine with either.

The main reason I decided to use the global is that `quit_program` really has little to do with an arbitrary given connection. Since `quit_program` is the final function doesn't return, it should work directly with global states, not some arbitrary parameter. Whether the global states are kept in global variables, or we write a class to keep track of global states and make `quit_program` and such instance methods, I don't care much at this point. (But that's certainly a good thing for unit testing purposes.)

Also, I was trying to handle ^D in `read_next_command`, which means embedding `quit_program`. We can't pass an arbitrary `conn` into `read_next_command`, that would look ridiculous.

> I'm thinking of supporting program options at omniprompt in future. Would add great flexibility.

Greater flexibility, sure. But there's still one global connection. I don't see how that would change.